lib/compilers/manual-solutions-compiler.ts

```diff
@@ -0,0 +1,67 @@
+import * as fs from "fs";
+import * as path from "path";
+import tmp from "tmp";
+import Course from "../models/course";
+import Language from "../models/language";
+import { glob } from "glob";
+
+class ManualSolutionsCompiler {
+  course: Course;
+
+  constructor(course: Course) {
+    this.course = course;
+  }
+
+  async compileAll() {
+    const languages = this.languages();
+
+    for (const language of languages) {
+      await this.compileForLanguage(language);
+    }
+  }
+
+  async compileForLanguage(language: Language) {
+    console.log(`- compiling manual solutions for ${this.course.slug}-${language.slug}`);
+
+    const stages = this.course.stages;
+    const firstStage = stages[0];
+    const firstStageCodeDirectory = path.join(this.course.solutionsDir, language.slug, firstStage.solutionDir, "code");
+
+    const firstStageFiles = await glob("**/*", {
+      cwd: firstStageCodeDirectory,
+      dot: true,
+      nodir: true,
+    });
+
+    for (let i = 1; i < stages.length - 1; i++) {
+      const currentStage = stages[i];
+      const currentStageCodeDirectory = path.join(this.course.solutionsDir, language.slug, currentStage.solutionDir, "code");
+
+      if (!fs.existsSync(currentStageCodeDirectory)) {
+        continue;
+      }
+
+      for (const file of firstStageFiles) {
+        if (!file.endsWith(language.codeFileExtension)) {
+          const sourcePath = path.join(firstStageCodeDirectory, file);
+          const targetPath = path.join(currentStageCodeDirectory, file);
+
+          if (fs.existsSync(sourcePath)) {
```

Comment:
`sourcePath` **must** exist at this point since it was fetched using `glob` above, can remove this check